### PR TITLE
test(NODE-3293): clean up change stream tests setImmediate usage

### PR DESCRIPTION
**Pull request by nbbeeken**
Change stream test clean up

### CODE REVIEW

File: test/functional/change_stream.test.js [hunk 1]
```diff
@@ -176,26 +171,25 @@ const pipeline = [
...
+    try {
+      await db.createCollection('test');
+    } catch {
+      // ns already exists, don't care
+    } finally {
+      await client.close();
+    }
   });
-  afterEach(() => mock.cleanup());
+  afterEach(async () => await mock.cleanup());
```
Comment (emadum) [inline test/functional/change_stream.test.js:192]: Just wondering, does switching to async/await in the hooks fix something or is this just stylistic?

File: test/functional/change_stream.test.js [hunk 2]
```diff
@@ -1792,109 +1786,86 @@ describe('Change Streams', function () {
...
-        clearInterval(repeatInsert);
-      }
+      let err;
       if (changeStream) {
-        await changeStream.close();
+        try {
+          if (changeStream[kMode] === 'emitter') {
+            // shutting down the client will end the session, if this happens before
+            // the stream initialization aggregate operation is processed, it will throw
+            // sa ession ended error, which can't be caught if we end the stream, so
```
Comment (emadum) [inline test/functional/change_stream.test.js:1809]:
```suggestion
            // a session ended error, which can't be caught if we end the stream, so
```

**Review by emadum — APPROVED** [review]
Typo fix and a question but this LGTM 👍

File: test/functional/change_stream.test.js (hunk 1 above)
Comment (nbbeeken) [reply test/functional/change_stream.test.js:192]: Stylistic mostly but when debugging I can step over this line without going into the mocha lib to see the result of the cleanup.

**Pull request merged**